virtual_rainforest/core/schema.py

```diff
@@ -0,0 +1,148 @@
+"""The :mod:`~virtual_rainforest.core.schema` module  provides the
+:func:`~virtual_rainforest.core.schema.load_schema` and
+:func:`~virtual_rainforest.core.schema.merge_schema` functions to build a single
+JSONSchema configuration validator from individual module schames. The JSONSchema
+validation is extended using the :func:`~virtual_rainforest.core.schema.set_defaults`
+iterator to provide a  global JSONSchema validator instance that handles
+the provision of default values within schema
+(:data:`~virtual_rainforest.core.registry.ValidatorWithDefaults`).
+"""  # noqa: D205, D415
+
+
+import json
+from pathlib import Path
+from typing import Any, Iterator
+
+import dpath.util  # type: ignore
+from jsonschema import Draft202012Validator, exceptions, validators
+
+from virtual_rainforest.core.logger import LOGGER
+
+
+def set_defaults(
+    validator: type[Draft202012Validator],
+    properties: dict[str, Any],
+    instance: dict[str, Any],
+    schema: dict[str, Any],
+) -> Iterator:
```

Comment:
What is this an `Iterator` of? Can the type be made more specific?

Reply:
It iterates over the iterates over the entries in a JSONSchema document - it isn't clear from the `python-jsonschema` docs what the types are.